cmd/juju/controller/destroy.go

```diff
@@ -282,10 +285,12 @@ upgrade the controller to version 2.3 or greater.
 			}
 		}
 		ctx.Infof("All hosted models reclaimed, cleaning up controller machines")
-		return environs.Destroy(controllerName, controllerEnviron, cloudCallCtx, store)
+		return environsDestroy(controllerName, controllerEnviron, cloudCallCtx, store)
 	}
 }
 
+var environsDestroy = environs.Destroy
```

Comment:
If we are making the change to be able to patch this (yay), it should be done by defining as an attribute on the command and passed in, like the controller credential api func.
Then we also avoid Patch in tests